types/leaflet/index.d.ts

```diff
@@ -587,37 +641,429 @@ export abstract class Evented extends Class {
+        'dragstart' | 'drag' | 'add' | 'remove' | 'loading' | 'error' | 'update' |
+        'down' | 'predrag',
+                            fn: LeafletEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'resize',
+                            fn: ResizeEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'popupopen' | 'popupclose',
+                            fn: PopupEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'tooltipopen' | 'tooltipclose',
+                            fn: TooltipEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'locationerror',
+                            fn: ErrorEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'locationfound',
+                            fn: LocationEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'click' | 'dblclick' | 'mousedown' | 'mouseup' | 'mouseover' |
+        'mouseout' | 'mousemove' | 'contextmenu' | 'preclick',
+                            fn: LeafletMouseEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'keypress' | 'keydown' | 'keyup',
+                            fn: LeafletKeyboardEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'zoomanim',
+                            fn: ZoomAnimEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'dragend',
+                            fn: DragEndEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'tileunload' | 'tileloadstart' | 'tileload' | 'tileabort',
+                            fn: TileEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: 'tileerror',
+                            fn: TileErrorEventHandlerFn, context?: any): this;
+    addOneTimeEventListener(type: string, fn: LeafletEventHandlerFn, context?: any): this;
+
+    /**
+     * Alias for once(...)
+     *
+     * Behaves as on(...), except the listener will only get fired once and then removed.
+     */
+    addOneTimeEventListener(eventMap: LeafletEventHandlerFnMap): this;
+    // tslint:enable:unified-signatures
+
+    /**
+     * Alias for fire(...)
+     *
+     * Fires an event of the specified type. You can optionally provide a data
+     * object — the first argument of the listener function will contain its properties.
+     * The event might can optionally be propagated to event parents.
+     */
+    fireEvent(type: string, data?: any, propagate?: boolean): this;
+
+    /**
+     * Alias for listens(...)
+     *
+     * Returns true if a particular event type has any listeners attached to it.
+     */
+    hasEventListeners(type: string): boolean;
+}
+
+// tslint:disable:strict-export-declare-modifiers
+declare class MixinType {
+    Events: Events;
+}
+
+export const Mixin: MixinType;
+
+/**
+ * Base class of Leaflet classes supporting events
+ */
+export abstract class Evented extends Class {
```

Comment:
You can add `interface Evented extends Event` to extend both Class and Events

Reply:
@navya9singh Can you elaborate? I tried variations of that and couldn't get it to work

Reply:
Here's an abstract example.

```ts
class A { a = 1 }
class B { b = 2 }
interface C extends A, B { }
abstract class C {
}
declare var c: C
c.a + c.b
```

I think you should be able to add `interface Evented extends Events {}` before line 767. If not, what is the error you see when you try?

Reply:
@sandersn I tried that, but with

```
interface Evented extends Events, Class {}
export abstract class Evented {}
```

I get:

```
ERROR: 825:33  expect  TypeScript@4.8 compile error: 
Property 'extend' does not exist on type 'typeof GridLayer'.
ERROR: 836:38  expect  TypeScript@4.8 compile error: 
Property 'extend' does not exist on type 'typeof GridLayer'.
```

Same with

```
interface Evented extends Events, Class {}
export abstract class Evented extends Events {}
```

With 

```
interface Evented extends Events, Class {}
export abstract class Evented extends Class {}
```

Self `npm test` passes, but not `npm run test-all`, with

```
rror in react-leaflet-markercluster/v2
Error: Errors in typescript@4.8 for external dependencies:
../../leaflet/index.d.ts(767,11): error TS2430: Interface 'Evented' incorrectly extends interface 'Events'.
  Types of property 'on' are incompatible.
    Type '{ (type: "spiderfied" | "unspiderfied", fn?: SpiderfyEventHandlerFn | undefined, context?: any): this; (type: "animationend", fn?: AnimationEndEventHandlerFn | undefined, context?: any): this; }' is not assignable to type '{ (type: "baselayerchange" | "overlayadd" | "overlayremove", fn: LayersControlEventHandlerFn, context?: any): this; (type: "layeradd" | "layerremove", fn: LayerEventHandlerFn, context?: any): this; (type: "add" | ... 18 more ... | "predrag", fn: LeafletEventHandlerFn, context?: any): this; (type: "resize", fn: Resiz...'.
      Types of parameters 'type' and 'type' are incompatible.
        Type '"baselayerchange" | "overlayadd" | "overlayremove"' is not assignable to type '"spiderfied" | "unspiderfied"'.
          Type '"baselayerchange"' is not assignable to type '"spiderfied" | "unspiderfied"'.

    at testTypesVersion (/home/hugo/projects/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:235:11)
    at runTests (/home/hugo/projects/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:176:5)
```

With similar errors from other packages

Reply:
Sounds like there's some complex merging happening when there's a separate `interface Evented` declaration. Without playing with it myself I'm not sure how hard it would be to fix. I'd recommend leaving it as-is personally.